abi/abi.go

```diff
@@ -16,22 +16,31 @@
 
 package abi
 
-import "github.com/DE-labtory/koa/crpyto"
+import (
+	"encoding/json"
+	"strings"
+)
 
-// The implementation below is implemented for the abi spec of ethereum.
-// https://solidity.readthedocs.io/en/develop/abi-spec.html
+type ABI struct {
+	Methods []Method
+}
 
-// Encode abi parameters
-func Encode(params ...[]byte) []byte {
-	return nil
+func New(abiJSON string) (ABI, error) {
+	reader := strings.NewReader(abiJSON)
+	dec := json.NewDecoder(reader)
+
+	var abi ABI
+	if err := dec.Decode(&abi); err != nil {
+		return ABI{}, err
```

Comment:
Can you return nil instead of ABI? (It is about return type)

Reply:
ok! there is no reason to return `ABI{}` instead of `nil` ... :)